core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -127,29 +127,39 @@ private static boolean isMasterChannelException(TransportException exp) {
     }
 
     /**
-     * Send a shard failed request to the master node to update the cluster state with the failure of a shard on another node.
+     * Convenience method for {@link #remoteShardFailed(ShardId, String, long, String, Exception, Listener)}.
+     */
+    public void remoteShardFailed(final ShardRouting shardRouting, long primaryTerm, final String message, @Nullable final Exception failure, Listener listener) {
```

Comment:
This method is only called once in production and once in test code.  I wonder if it makes sense to get rid of it in favor of just using the method below, in order to simplify `ShardStateAction`'s API:

```public void remoteShardFailed(final ShardId shardId, String allocationId, long primaryTerm, final String message, @Nullable final Exception failure, Listener listener)```